cvat/apps/engine/views.py

```diff
@@ -147,6 +151,18 @@ def share(request):
             return Response("{} is an invalid directory".format(param),
                 status=status.HTTP_400_BAD_REQUEST)
 
+    @staticmethod
+    @action(detail=False, methods=['GET'], serializer_class=AnnotationFormatSerializer)
+    def formats(request):
+        response = {
+            'upload': [p.name for p in AnnoParser.objects.all()],
+            'download': [d.name for d in AnnoDumper.objects.all()],
+        }
+
+        serializer = AnnotationFormatSerializer(data=response)
```

Comment:
- You serializer should be able to serialise DB objects without your extra code.
- Use single naming convention inside your code: Anno -> Annotation, upload -> parser, download -> dumper.
- I don't like `/server/formats` but let me see its output after you fix first comment.